src/authentication/middleware/index.ts

```diff
@@ -0,0 +1,31 @@
+import { NextFunction, Request, Response } from "express";
+import activeFeature from "../../feature.flag";
+import logger from "../../logger";
+import * as pageURLs from "../../model/page.urls";
+
+export default (req: Request, res: Response, next: NextFunction) => {
+
+  logger.debug("Check if user is authenticated");
+
+  if (!req.chSession.isSignedIn()) {
+
+    logger.debug("User not authenticated - redirecting to login screen");
+
+    let returnToUrl: string = pageURLs.PROMISE_TO_FILE;
+
+    const referringPageURL = req.header("Referer") as string;
+
+    // Redirect the user to to the start screen if they weren't already on the PTF journey
```

Comment:
what happens if they are signed in but the referer is blank?

Reply:
As discussed, it's unlikely to be a problem.

Reply:
Seems that user is allowed to jump into the journey at that point. It's potentially something we want to address but this is the same behaviour as Extensions, so probably not urgent.